test/functional/p2p_getaddr_caching.py

```diff
@@ -60,16 +69,28 @@ def run_test(self):
         N = 5
         cur_mock_time = int(time.time())
         for i in range(N):
-            addr_receiver = self.nodes[0].add_p2p_connection(AddrReceiver())
-            addr_receiver.send_and_ping(msg_getaddr())
+            addr_receiver_local = self.nodes[0].add_p2p_connection(AddrReceiver())
+            addr_receiver_local.send_and_ping(msg_getaddr())
+            addr_receiver_onion1 = self.nodes[0].add_p2p_connection(AddrReceiver(), dstport=self.onion_port1)
+            addr_receiver_onion1.send_and_ping(msg_getaddr())
+            addr_receiver_onion2 = self.nodes[0].add_p2p_connection(AddrReceiver(), dstport=self.onion_port2)
+            addr_receiver_onion2.send_and_ping(msg_getaddr())
+
             # Trigger response
             cur_mock_time += 5 * 60
             self.nodes[0].setmocktime(cur_mock_time)
-            addr_receiver.wait_until(addr_receiver.addr_received)
-            responses.append(addr_receiver.get_received_addrs())
-        for response in responses[1:]:
-            assert_equal(response, responses[0])
-        assert(len(response) == MAX_ADDR_TO_SEND)
+            addr_receiver_local.wait_until(addr_receiver_local.addr_received)
+            addr_receiver_onion1.wait_until(addr_receiver_onion1.addr_received)
+            addr_receiver_onion2.wait_until(addr_receiver_onion2.addr_received)
+            responses.append(addr_receiver_local.get_received_addrs())
+            responses.append(addr_receiver_onion1.get_received_addrs())
+            responses.append(addr_receiver_onion2.get_received_addrs())
+
+        for i in range(0, len(responses) - 3):
+            assert(responses[i] != responses[i+1])
```

Comment:
This indexing is hard to read. Perhaps you could try using named variables instead?

Reply:
Got rid of the `responses` list entirely in favor of simply comparing to the last received response.